src/main/java/uk/gov/justice/hmpps/prison/repository/jpa/repository/CourtEventChargeRepository.java

```diff
@@ -14,7 +14,8 @@ public interface CourtEventChargeRepository extends CrudRepository<CourtEventCha
         "join cec.eventAndCharge.offenderCharge oc " +
         "join oc.offenderBooking ob " +
         "join ob.offender o " +
-        "where o.nomsId = :offenderId"
+        "where o.nomsId = :offenderId and " +
+        "exists (select id from CourtEvent where id = cec.eventAndCharge.courtEvent.id)"
```

Comment:
Would be good to have a comment here to explain what is going on - as someone in future will just look at the line and think it isn't doing anything since the court event must exist.
Is there an existing repository test for this class - would be good to have a test for this too...

Reply:
I'll add a comment here. 
I looked at adding a test but it was difficult to create the invalid data because the constraint exists in the schema we run tests against.